HPCC-32118 Default to case-insensitive metric filtering

GordonSmith opened the pull request:
Introduced an option to toggle between case-sensitive and case-insensitive filtering

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [ ] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [ ] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

jeclrsg commented on esp/src/src-react/components/Metrics.tsx:
```diff
@@ -67,8 +67,12 @@ class TableEx extends Table {
-                    return true;
+                if (row[key]?.indexOf) {
+                    const value = (!matchCase && row[key]?.toLowerCase) ? row[key]?.toLowerCase() : row[key];
+                    const filterValue = (!matchCase && row[key]?.toLowerCase) ? filter.toLowerCase() : filter;
```
should this be `!matchCase && filter?.toLowerCase`?

GordonSmith replied:
No this is correct - the idea is to only lower the filterValue (which is always a string) if the value had been lowered

GordonSmith replied:
Updated logic

jeclrsg reviewed (changes requested):
@GordonSmith One question about a conditional statement. Looked good otherwise

GordonSmith committed:
HPCC-32118 Default to case-insensitive metric filtering

Introduced an option to toggle between case-sensitive and case-insensitive filtering

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

jeclrsg commented on esp/src/src-react/components/Metrics.tsx:
```diff
@@ -625,7 +629,12 @@ export const Metrics: React.FunctionComponent<MetricsProps> = ({
+                                <Stack.Item grow>
+                                    <SearchBox value={scopeFilter} onChange={onChangeScopeFilter} iconProps={filterIcon} placeholder={nlsHPCC.Filter} />
+                                </Stack.Item>
+                                <ToggleButton appearance="subtle" icon={<TextCaseTitleRegular />} title={nlsHPCC.MatchCase} checked={matchCase} onClick={() => { setMatchCase(!matchCase); }} />
```
Only thing I noticed when I checked this branch out is the state of this button is pretty subtle. I thought maybe `icon={matchCase ? <TextCaseTitleRegular /> : <TextCaseLowercaseRegular />}`?

jeclrsg reviewed (commented):
@GordonSmith Only one comment about how that match case button actually shows up in the UI. Seems pretty subtle as to what the status of it is

GordonSmith commented:
Rebased for pull
